src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java

```diff
@@ -626,7 +633,9 @@ public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> r
 
         for (int i = 0; i < numOfRubricSubQuestions; i++) {
```

Comment:
We can extract this `for` loop to a common method that takes in `responses` as parameters. We can pass `responses` and `responses excluding self` and u will get two different rendered HTML table.

The main idea is to pass different responses to the methods and reuse the logic. This aligns with the meaning of `statistics that excluding self` -- statistics generated if self-evaluation is not done at all.

BTW, you can move
```
....
RubricStatistics statistics = new RubricStatistics(responsesForStatistics, fqd);
....
```
inside the common method also.

If u follow my suggestion above, you also don't need to change anything inside `RubricStatistics`

Reply:
That is great advice :)
> If u follow my suggestion above, you also don't need to change anything inside RubricStatistics

I have made the logic simpler now, but I dont get what you mean by this. The current changes will still be needed in RubricStatistics for calculating the responses for excluding self

Reply:
You can have a separate method to filter out all self responses.
``` java
responses = xxxx
responsesWithoutSelf = filterSelfResponses(responses)
responsesTable = getQuestionResultsStatisticsBodyHtml(responses)
responsesWithoutSelfTable = getQuestionResultsStatisticsBodyHtml(responsesWithoutSelf)
```
BTW, don't use my variable naming :P I believe u could give better name.

Reply:
I had extracted the `for(...)` loop as suggested but instead of passing `responses[][]` I decided to pass  `FeedbackRubricQuestionDetails fqd, List<FeedbackResponseAttributes> responsesForStatistics` because these are used for some `if()` checks inside the for loop.

https://github.com/TEAMMATES/teammates/pull/8517/commits/a87df826c03ae14cdee7402cc87c4b0ffd82eaf0 You can refer to this commit for the changes.

I feel this helps in separating logic of rendering tables and getting statistics.
Moreover, the code is pretty much simplified in `ResultStatistics`.

Lmk if the above commit has simplified things as per your request